src/vs/gitpod/browser/workbench/workbench.html

```diff
@@ -3,6 +3,72 @@
 <html>
 	<head>
 		<script>
+			async function gitpodMetricsAddCounter(metricsName, labels, value) {
+				function getMetricsUrl() {
+					const baseWorkspaceIDRegex = '(([a-f][0-9a-f]{7}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})|([0-9a-z]{2,16}-[0-9a-z]{2,16}-[0-9a-z]{8,11}))';
+					// this pattern matches URL prefixes of workspaces
+					const workspaceUrlPrefixRegex = RegExp(`^([0-9]{4,6}-)?${baseWorkspaceIDRegex}\\.`);
+					const url = new URL(window.location.href);
+					url.search = '';
+					url.hash = '';
+					url.pathname = '';
+					if (url.host.match(workspaceUrlPrefixRegex)) {
+						url.host = url.host.split('.').splice(2).join('.');
+					} else {
+						return;
+					}
+					const hostSegments = url.host.split('.');
+					if (hostSegments[0] !== 'ide') {
+						url.host = 'ide.' + url.host;
+					}
+					url.pathname = '/metrics-api';
+					return url.toString();
+				}
+				try {
+					const metricsUrl = getMetricsUrl();
+					if (!metricsUrl) {
+						return false;
+					}
+					const url = `${metricsUrl}/metrics/counter/add/${metricsName}`;
+					const params = { value, labels };
+					const response = await fetch(url, {
+						method: 'POST',
+						body: JSON.stringify(params),
+						headers: { 'Content-Type': 'application/json;' },
+						credentials: 'omit',
+					});
+					if (!response.ok) {
+						const data = await response.json(); // { code: number; message: string; }
+						console.error(`Cannot report metrics with addCounter: ${response.status} ${response.statusText}`, data);
+						return false;
+					}
+					return true;
+				} catch (err) {
+					console.error('Cannot report metrics with addCounter, error:', err);
+					return false;
+				}
+			}
+
+			const onVsCodeWorkbenchError = (event) => {
+				if (typeof event?.target?.getAttribute !== 'function') {
+					gitpodMetricsAddCounter('gitpod_supervisor_frontend_error_total').then().catch((e) => {});
+					return;
+				}
+				const labels = new Map();
+
+				// We take a look at what is the resource that was attempted to load;
+				const resourceSource = event.target.getAttribute('src') || event.target.getAttribute('href');
+
+				// If the event has a `target`, it means that it wasn't a script error
+				if (resourceSource) {
+					labels.set('resource', 'vscode-web-workbench');
+					labels.set('error', 'LoadError');
+				} else {
+					labels.set('error', 'Unknown');
+				}
+				gitpodMetricsAddCounter('gitpod_supervisor_frontend_error_total', labels).then().catch((e) => {});
```

Comment:
please see https://github.com/gitpod-io/openvscode-server/pull/417#discussion_r955425089

```suggestion
				gitpodMetricsAddCounter('gitpod_supervisor_frontend_error_total', labels);
```